R/spliceHelperFunctions.R

```diff
@@ -96,232 +96,180 @@ findSpliceOverlapsByDist <-function(query, subject, ignore.strand=FALSE, maxDist
+                       end.last.subject = end(unlist(subjectEndRng)),
+                       strand = strand)  
+  tempTable <- tempTable %>% mutate(alternativeFirstExon=!ifelse(strand!='-',
+                                                                start.first.query <= end.first.subject & end.first.query >= start.first.subject,
+                                                                end.first.query >= start.first.subject & start.first.query <= end.first.subject),
+                                   alternativeLastExon=!ifelse(strand!='-',
+                                                               end.last.query >= start.last.subject & start.last.query <= end.last.subject,
+                                                               start.last.query <= end.last.subject & end.last.query >= start.last.subject),
+                                   alternativeTSS=ifelse(strand!='-',
+                                                         start.first.subject-start.first.query,
+                                                         end.first.query- end.first.subject )*!alternativeFirstExon,
+                                   alternativeTES=ifelse(strand!='-', 
+                                                         end.last.query-end.last.subject,
+                                                         start.last.subject-start.last.query)*!alternativeLastExon)
+  tempTable <- tempTable %>% select(alternativeFirstExon, alternativeLastExon, alternativeTSS, alternativeTES)
+  #internal start/end query 
+  queryTable <- annotateInternalStartEnd(queryStartRng, queryEndRng, subjectFullRng, tempTable)
+  colnames(queryTable) <- paste(colnames(queryTable),"query",sep=".")
+  #internal start/end subject
+  subjectTable <- annotateInternalStartEnd(subjectStartRng, subjectEndRng, queryFullRng, tempTable)
+  colnames(subjectTable) <- paste(colnames(subjectTable),"subject",sep=".")
+  tempTable <- cbind(tempTable, queryTable, subjectTable)
+  return (tempTable)
+}
+
+#' annotate internal start and end first exons                                    
+#' @noRd
+annotateInternalStartEnd <- function(StartRng, EndRng, FullRng, alterEndTable){
+  Start.Full.Rng <- expandMatchedRanges(StartRng, FullRng) 
+  End.Full.Rng <- expandMatchedRanges(EndRng, FullRng)
+  #internal start
+  ExonIntersectStart <- pintersect(Start.Full.Rng,mcols(Start.Full.Rng)$matchRng, resolve.empty='start.x')
+  #internal end
+  ExonIntersectEnd=pintersect(End.Full.Rng,mcols(End.Full.Rng)$matchRng, resolve.empty='start.x')
+  tempTable <- tibble(internalFirstExon = tapply(width(ExonIntersectStart), mcols(Start.Full.Rng)$IdMap, sum)!= 0 & alterEndTable$alternativeFirstExon,
+                      internalLastExon = tapply(width(ExonIntersectEnd), mcols(End.Full.Rng)$IdMap, sum)!=0 & alterEndTable$alternativeLastExon)
+  return(tempTable)
+}
+
+#' annotate intron retention 
+#' @noRd
+annotateIntronRetent <- function(Splice.FullSplice.Rng, annotatedTable){
```

Comment:
is the annotatedTable required here?

Reply:
Thanks for the reminder! It is not required.